File: Python Programs/unit2/StackLinkedList.py
from __future__ import print_function

top = None


class Node(object):

    def __init__(self, data=None, prev_node=None):
        self.data = data
        self.prev_node = prev_node


def pushS():
    global top
    print("Enter the value:")
    value = int(input())
    newNode = Node(data=value)
    if top is None:
        top = newNode
    else:
        newNode.prev_node = top
        top = newNode


def popS():
    global top
    temp = top.data
    top = top.prev_node
    return temp


def display():
    global top
    if top is None:
        print("Stack is empty:")
    else:
        temp = top
        while temp is not None:
            print(str(temp.data) + " ", end="")
            temp = temp.prev_node
        print("")


while True:
    print("Stack implementation in Linked List")
    print("1.Push\n2.Pop\n3.Display\n4.Exit")
    c = int(input("Enter your choice : "))
    if c == 1:
        pushS()
    elif c == 2:
        print("The element popped is %d" % popS())
    elif c == 3:
        display()
    elif c == 4:
        exit(0)

    else:
        print("Wrong Choice")

File: Java Programs/Chapter 2/Stack_LinkedList/src/stack_linkedlist/Stack_LinkedList.java

package stack_linkedlist;

//Program to implement stack using array.
//Maximum size of the stack is 20.
//Elements of any datatype can be inserted into the stack.
import java.util.*;

//Program to implement Stack using linkedlist.
//This program will not work if inputs are not given properly.
//The data to be pushed or popped into and out of the stack may be of any datatype.
import java.io.*;
import java.util.*;
class Node
{
	protected Object value;
	protected Node next_address;
	public Node()
	{
		value=null;
		next_address=null;
	}
	public Node(Object element,Node top)
	{
		value=element;
		next_address=top;
	}
	public Object getValue()
	{
		return value;
	}
	public Node getNext_address()
	{
		return next_address;
	}

}
class Stack_link
{
	protected Node top;
	protected Node newNode;
	public Stack_link()
	{
		top=null;
	}
	public void push(Object element)
	{
		newNode=new Node(element,top);
		if(newNode==null)
		{
			System.out.println("Sufficient Memory Not Available to Create Nodes");
		}
		else
		{
			top=newNode;
		}
	}
	public Object pop()
	{
		Object element;
		if(isEmpty())
		{
			System.out.println("Stack is empty");
			element=null;
		
		}
		else
		{
			
			element=top.getValue();
			top=top.getNext_address();
		}
		return element;
	}
	public boolean isEmpty()
	{
		if(top==null)
		{
			return true;
		}
		else
		{
			return false;
		}
	}
	public void displayStack()
	{
		Node temp=top;
		System.out.print("Stack: ");
		while(temp!=null)
		{
			System.out.print(temp.getValue()+" |");
			temp=temp.getNext_address();
		}
	}
	
}
public class Stack_LinkedList
{
	public static void main(String args[]) throws Exception
	{
		Scanner oin=new Scanner(System.in);
		Stack_link stk=new Stack_link();
		Object element;
		int choice;
		
		while(true)
		{
			
			System.out.println("\n\nEnter the operation to be performed\n 1.Push\n 2.Pop\n 3.Exit");
			System.out.print("Option: ");
			String s1=oin.nextLine();
			choice=Integer.parseInt(s1);
			if(choice==1)
			{
					System.out.print("Enter the element to be pushed: ");
					String s=oin.nextLine();
					Object data=(Object)s;
					stk.push(data);
					stk.displayStack();
			}
			else if(choice==2)
			{
				element= stk.pop();
				if(element!=null)
				{
					System.out.println("Poppped Element: "+element);
				}
				stk.displayStack();
			}
			else if(choice==3)
			{
				break;
			}
			else
			{
				System.out.println("Invalid Choice");
			}
				
		}
	}
	
}